scrapy/http/request/__init__.py

```diff
@@ -4,17 +4,36 @@
 
 See documentation in docs/topics/request-response.rst
 """
+import inspect
+from typing import Optional, Tuple
+
 from w3lib.url import safe_url_string
 
+import scrapy
+from scrapy.http.common import obsolete_setter
 from scrapy.http.headers import Headers
-from scrapy.utils.python import to_bytes
+from scrapy.utils.curl import curl_to_request_kwargs
+from scrapy.utils.python import to_bytes, to_unicode
 from scrapy.utils.trackref import object_ref
 from scrapy.utils.url import escape_ajax
-from scrapy.http.common import obsolete_setter
-from scrapy.utils.curl import curl_to_request_kwargs
 
 
 class Request(object_ref):
+    """Represents an HTTP request, which is usually generated in a Spider and
+    executed by the Downloader, thus generating a :class:`Response`.
+    """
+
+    attributes: Tuple[str, ...] = (
+        "url", "method", "headers", "body", "cookies", "meta", "flags",
+        "encoding", "priority", "dont_filter", "callback", "errback", "cb_kwargs",
```

Comment:
A nitpick: would it make sense to use the same order as in `__init__`, to make it easier to ensure they match?